Description: Correctly mark unmanaged devices
 Add interface (iface ...) to well_known_interface no matter if we find a
 valid connection configuration or not. This way interfaces are correctly
 marked unmanaged for configurations NM does not understand.
Author: Michael Biebl <biebl@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=569215
Index: network-manager/system-settings/plugins/ifupdown/plugin.c
===================================================================
--- network-manager.orig/system-settings/plugins/ifupdown/plugin.c	2010-12-05 04:24:58.153750890 +0100
+++ network-manager/system-settings/plugins/ifupdown/plugin.c	2010-12-05 04:27:03.157754436 +0100
@@ -368,6 +368,7 @@
 			/* Remove any connection for this block that was previously found */
 			exported = g_hash_table_lookup (priv->iface_connections, block->name);
 			if (exported) {
+				PLUGIN_PRINT("SCPlugin-Ifupdown", "deleting %s from iface_connections", block->name);
 				nm_settings_connection_interface_delete (NM_SETTINGS_CONNECTION_INTERFACE (exported),
 				                                         ignore_cb,
 				                                         NULL);
@@ -377,11 +378,14 @@
 			/* add the new connection */
 			exported = nm_ifupdown_connection_new (block);
 			if (exported) {
+				PLUGIN_PRINT("SCPlugin-Ifupdown", "adding %s to iface_connections", block->name);
 				g_hash_table_insert (priv->iface_connections, block->name, exported);
-				g_hash_table_insert (priv->well_known_interfaces, block->name, "known");
 			}
+			PLUGIN_PRINT("SCPlugin-Ifupdown", "adding iface %s to well_known_interfaces", block->name);
+			g_hash_table_insert (priv->well_known_interfaces, block->name, "known");
 		} else if (!strcmp ("mapping", block->type)) {
 			g_hash_table_insert (priv->well_known_interfaces, block->name, "known");
+			PLUGIN_PRINT("SCPlugin-Ifupdown", "adding mapping %s to well_known_interfaces", block->name);
 		}
 		block = block->next;
 	}
